### PR TITLE
Fix/segments can not be deleted when archived

### PR DESCRIPTION
**Pull request by FredrikOseberg**
This PR makes a change to the logic of segments: 

(1) You can now delete segments that are being referenced in strategies on archived toggles
(2) We will no longer show "used in project" and "used in feature" when the toggle referencing it is archived

Skjermbilde 2024-02-29 kl  13 58 55

Also added a warning when deleting: 
Skjermbilde 2024-02-29 kl  13 59 21

Discussion points: 
* I have added tests to verify the behavior, and we should only be touching the usage metrics with these added queries. But I see that one of the methods that calls getAll is used in the proxy repository, which makes me wary of making this change.

I've tried to map out the impact of my changes here:
Skjermbilde 2024-02-29 kl  14 21 06

### CODE REVIEW

File: src/lib/db/segment-store.ts [hunk 1]
```diff
@@ -174,7 +175,13 @@ export default class SegmentStore implements ISegmentStore {
                     T.featureStrategies,
                     `${T.featureStrategies}.id`,
                     `${T.featureStrategySegment}.feature_strategy_id`,
-                );
+                )
+                .leftJoin(
```
Comment (FredrikOseberg) [inline src/lib/db/segment-store.ts:179]: Here, the idea is to not returnstrategies if the archived at is null.

**Review by sjaanus — APPROVED** [review]
Nice!

File: src/lib/db/segment-store.ts (hunk 1 above)
Comment (FredrikOseberg) [inline src/lib/db/segment-store.ts:179]: This should only affect usage count

File: src/lib/db/segment-store.ts [hunk 2]
```diff
@@ -229,11 +236,15 @@ export default class SegmentStore implements ISegmentStore {
                     'used_in_projects',
                     'used_in_features',
                 )
-                .countDistinct(
-                    `${T.featureStrategies}.project_name AS used_in_projects`,
+                .select(
```
Comment (FredrikOseberg) [inline src/lib/db/segment-store.ts:239]: Should only affect usage count

File: src/lib/features/feature-toggle/feature-toggle-strategies-store.ts [hunk 3]
```diff
@@ -814,7 +814,14 @@ class FeatureStrategiesStore implements IFeatureStrategiesStore {
                 `${T.featureStrategySegment}.feature_strategy_id`,
                 `${T.featureStrategies}.id`,
             )
-            .where(`${T.featureStrategySegment}.segment_id`, '=', segmentId);
+            .join(
```
Comment (FredrikOseberg) [inline src/lib/features/feature-toggle/feature-toggle-strategies-store.ts:817]: Here, we don't want to return strategies that exist on archived toggles

**Pull request closed without merging**